ShaiRandom/Generators/IEnhancedRandomExtensions.cs

```diff
@@ -539,5 +538,30 @@ public static void SetWith(this IEnhancedRandom rng, IEnhancedRandom other)
                 rng.SetSelectedState(i, 0xFFFFFFFFFFFFFFFFUL);
             }
         }
+
+        /// <summary>
+        /// Given two IEnhancedRandom objects that could have the same or different classes,
+        /// this returns true if they have the same class and same state, or false otherwise.
+        /// </summary>
+        /// <remarks>
+        /// Both of the arguments should implement <see cref="IEnhancedRandom.SelectState(int)"/>, or this
+        /// will throw an exception. This can be useful for comparing IEnhancedRandom classes.
+        /// </remarks>
+        /// <returns>true if the two objects have the same class and state, or false otherwise</returns>
+        public static bool Matches(this IEnhancedRandom left, IEnhancedRandom right)
```

Comment:
I definitely like the naming scheme better now that it is less confuse-able with IsEqual().